tests/queries/0_stateless/02932_kill_query_sleep.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+
+function wait_query_started()
+{
+    local query_id="$1"
+    while [[ $($CLICKHOUSE_CLIENT --query="SELECT count() FROM system.query_log WHERE query_id='$query_id' AND current_database = currentDatabase()") == 0 ]]; do
+        sleep 0.1;
+        $CLICKHOUSE_CLIENT --query "SYSTEM FLUSH LOGS;"
+    done
+}
+
+function kill_query()
+{
+    local query_id="$1"
+    $CLICKHOUSE_CLIENT --query "KILL QUERY WHERE query_id='$query_id'" >/dev/null
+    while [[ $($CLICKHOUSE_CLIENT --query="SELECT count() FROM system.processes WHERE query_id='$query_id'") != 0 ]]; do sleep 0.1; done
+}
+
+
+sleep_query_id="sleep_query_id_02932_kill_query_sleep_${CLICKHOUSE_DATABASE}_$RANDOM"
+
+# This sleep query wants to sleep for 1000 seconds (which is too long).
+# We're going to cancel this query later.
+sleep_query="SELECT sleep(1000)"
```

Comment:
It may break Hung Check in Stress Tests: https://s3.amazonaws.com/clickhouse-test-reports/58507/82d3d570530ebd014717ba0e11bfd975fe2502e7/stress_test__asan_.html

```
Found hung queries in processlist:
Row 1:
──────
is_initial_query:     1
user:                 default
query_id:             sleep_query_id_02932_kill_query_sleep_test_znckl0ze_20875
address:              ::1
port:                 48648
initial_user:         default
initial_query_id:     sleep_query_id_02932_kill_query_sleep_test_znckl0ze_20875
initial_address:      ::1
initial_port:         48648
interface:            1
os_user:              
client_hostname:      42f105a870e2
client_name:          ClickHouse client
client_revision:      54466
client_version_major: 24
client_version_minor: 1
client_version_patch: 1
http_method:          0
http_user_agent:      
http_referer:         
forwarded_for:        
quota_key:            
distributed_depth:    0
elapsed:              995.390784
is_cancelled:         0
is_all_data_sent:     0
read_rows:            1
read_bytes:           1
total_rows_approx:    0
written_rows:         0
written_bytes:        0
memory_usage:         0
peak_memory_usage:    0
query:                SELECT sleep(1000)
query_kind:           Select
thread_ids:           [34405,21823]
```

Reply:
I'll change the test.

Reply:
https://github.com/ClickHouse/ClickHouse/pull/58560